File: library/src/main/java/com/getstream/sdk/chat/rest/core/ChatChannelEventHandler.java
package com.getstream.sdk.chat.rest.core;

import com.getstream.sdk.chat.R;
import com.getstream.sdk.chat.StreamChat;
import com.getstream.sdk.chat.model.Event;
import com.getstream.sdk.chat.storage.Sync;

public abstract class ChatChannelEventHandler {
    public void onAnyEvent(Event event) {
    }

    public void onTypingStart(Event event) {
    }

    public void onTypingStop(Event event) {
    }

    public void onMessageNew(Event event) {
    }

    public void onMessageUpdated(Event event) {
    }

    public void onMessageDeleted(Event event) {
    }

    public void onMessageRead(Event event) {
    }

    public void onReactionNew(Event event) {
    }

    public void onReactionDeleted(Event event) {
    }

    public void onMemberAdded(Event event) {
    }

    public void onMemberRemoved(Event event) {
    }

    public void onMemberUpdated(Event event) {
    }

    public void onChannelUpdated(Event event) {
    }

    public void onChannelHidden(Event event) {
    }

    public void onChannelDeleted(Event event) {
    }

    public void onUserWatchingStart(Event event) {
    }

    public void onUserWatchingStop(Event event) {
    }

    public final void dispatchEvent(Event event) {
        onAnyEvent(event);
        switch (event.getType()) {
            case TYPING_START:
                onTypingStart(event);
                break;
            case TYPING_STOP:
                onTypingStop(event);
                break;
            case MESSAGE_NEW:
                event.getMessage().setSyncStatus(Sync.SYNCED);
                onMessageNew(event);
                break;
            case MESSAGE_UPDATED:
                event.getMessage().setSyncStatus(Sync.SYNCED);
                onMessageUpdated(event);
                break;
            case MESSAGE_DELETED:
                event.getMessage().setSyncStatus(Sync.SYNCED);
                event.getMessage().setText(StreamChat.getStrings().get(R.string.stream_delete_message));
                onMessageDeleted(event);
                break;
            case MESSAGE_READ:
                onMessageRead(event);
                break;
            case REACTION_NEW:
                event.getMessage().setSyncStatus(Sync.SYNCED);
                onReactionNew(event);
                break;
            case REACTION_DELETED:
                event.getMessage().setSyncStatus(Sync.SYNCED);
                onReactionDeleted(event);
                break;
            case MEMBER_ADDED:
                onMemberAdded(event);
                break;
            case MEMBER_REMOVED:
                onMemberRemoved(event);
                break;
            case MEMBER_UPDATED:
                onMemberUpdated(event);
                break;
            case CHANNEL_UPDATED:
                onChannelUpdated(event);
                break;
            case CHANNEL_HIDDEN:
                onChannelHidden(event);
                break;
            case CHANNEL_DELETED:
                onChannelDeleted(event);
                break;
            case USER_WATCHING_START:
                onUserWatchingStart(event);
                break;
            case USER_WATCHING_STOP:
                onUserWatchingStop(event);
                break;
        }
    }
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/ChatChannelEventHandler.kt
package io.getstream.chat.android.core.poc.library

import io.getstream.chat.android.core.poc.library.events.ChatEvent


abstract class ChatChannelEventHandler {
    fun onAnyEvent(event: ChatEvent) {}
    fun onTypingStart(event: ChatEvent) {}
    fun onTypingStop(event: ChatEvent) {}
    fun onMessageNew(event: ChatEvent) {}
    fun onMessageUpdated(event: ChatEvent) {}
    fun onMessageDeleted(event: ChatEvent) {}
    fun onMessageRead(event: ChatEvent) {}
    fun onReactionNew(event: ChatEvent) {}
    fun onReactionDeleted(event: ChatEvent) {}
    fun onMemberAdded(event: ChatEvent) {}
    fun onMemberRemoved(event: ChatEvent) {}
    fun onMemberUpdated(event: ChatEvent) {}
    fun onChannelUpdated(event: ChatEvent) {}
    fun onChannelHidden(event: ChatEvent) {}
    fun onChannelDeleted(event: ChatEvent) {}
    fun onUserWatchingStart(event: ChatEvent) {}
    fun onUserWatchingStop(event: ChatEvent) {}
    fun dispatchEvent(event: ChatEvent) {
        onAnyEvent(event)
        when (event.getType()) {
            EventType.TYPING_START -> onTypingStart(event)
            EventType.TYPING_STOP -> onTypingStop(event)
            EventType.MESSAGE_NEW -> {
                onMessageNew(event)
            }
            EventType.MESSAGE_UPDATED -> {
                onMessageUpdated(event)
            }
            EventType.MESSAGE_DELETED -> {
                //TODO: define R.string.stream_delete_message
                //event.message?.text = "Deleted message"
                //event.message?.text = StreamChat.getStrings().get(R.string.stream_delete_message)
                onMessageDeleted(event)

            }
            EventType.MESSAGE_READ -> onMessageRead(event)
            EventType.REACTION_NEW -> {
                onReactionNew(event)
            }
            EventType.REACTION_DELETED -> {
                onReactionDeleted(event)
            }
            EventType.MEMBER_ADDED -> onMemberAdded(event)
            EventType.MEMBER_REMOVED -> onMemberRemoved(
                event
            )
            EventType.MEMBER_UPDATED -> onMemberUpdated(
                event
            )
            EventType.CHANNEL_UPDATED -> onChannelUpdated(
                event
            )
            EventType.CHANNEL_HIDDEN -> onChannelHidden(
                event
            )
            EventType.CHANNEL_DELETED -> onChannelDeleted(
                event
            )
            EventType.USER_WATCHING_START -> onUserWatchingStart(
                event
            )
            EventType.USER_WATCHING_STOP -> onUserWatchingStop(
                event
            )
        }
    }
}
